examples/js/archive.js

```diff
@@ -281,87 +280,121 @@ function isJsonDetected(url) {
-      if (imgObjCollection.length > 1) {
-        let imageURL;
-        let options;
+    const imageCollectionObj = await map.imgGroup.recreateImagesFromJsonUrl(
+      jsonDownloadURL
+    );
+    let imgObjCollection = imageCollectionObj.imgCollectionProps;
+    // const avg_cm_per_pixel = imageCollectionObj.avg_cm_per_pixel; // this is made available here for future use. can't be used with legacy json files from mapknitter.org
 
-        const cornerBounds = getCornerBounds(imgObjCollection); 
-        map.fitBounds(cornerBounds);
-    
-        imgObjCollection.forEach((imageObj) => {
-          imageURL = extractImageSource(imageObj.src);
-          if (imageObj.nodes.length > 0) {
-            options = {
-              tooltipText: imageObj.tooltipText,
-              corners: imageObj.nodes,
-            };
-            placeImage(imageURL, options, false);
-          }
-        });
+    // for json file with single image object
+    if (Array.isArray(imgObjCollection[0])) {
+      imgObjCollection = updateLegacyJson(imgObjCollection[0]).collection;
+    } else {
+      // for json file with multiple image objects
+      imgObjCollection = updateLegacyJson(imgObjCollection).collection;
+    }
 
-        return;
-      }
+    // creates multiple images - this applies where multiple images are to be reconstructed
+    if (imgObjCollection.length > 1) {
+      let imageURL;
+      let options;
 
-      // creates single image - this applies where only one image is to be reconstructed
-      const cornerBounds = getCornerBounds(imgObjCollection); 
+      const cornerBounds = getCornerBounds(imgObjCollection);
       map.fitBounds(cornerBounds);
-      
-      const imageObj = (imgObjCollection[0])[0];
-      const imageURL = extractImageSource(imageObj.src);
-
-      if (imageObj.nodes.length) {
-        const options = {
-          tooltipText: imageObj.tooltipText,
-          corners: imageObj.nodes,
+
+      imgObjCollection.forEach((imageObj) => {
+        imageURL = extractImageSource(imageObj.src);
+        if (imageObj.nodes.length > 0) {
+          options = {
+            tooltipText: imageObj.tooltipText,
+            corners: imageObj.nodes,
+          };
+          placeImage(imageURL, options, false);
         }
-        placeImage(imageURL, options, false);
-      }
+      });
+
+      return;
+    }
+
+    // creates single image - this applies where only one image is to be reconstructed
+    const cornerBounds = getCornerBounds(imgObjCollection);
+    map.fitBounds(cornerBounds);
+
+    const imageObj = imgObjCollection[0][0];
+    const imageURL = extractImageSource(imageObj.src);
+
+    if (imageObj.nodes.length) {
+      const options = {
+        tooltipText: imageObj.tooltipText,
+        corners: imageObj.nodes,
+      };
+      placeImage(imageURL, options, false);
+    }
   }
-}
+
+  // creates multiple images - this applies where multiple images are to be reconstructed
```

Comment:
So this until line 398 - is it part of your fixes, or is this code you're not sure where it came from?

Reply:
Yea, this until 398 is part of my fixes